docs/repo-docs/guides/migrating-from-nx.mdx

```diff
@@ -0,0 +1,481 @@
+{
+  "packageManager": "yarn@1.22.19"
+}
+```
+
+</Tab>
+
+<Tab>
+
+```json title="./package.json"
+{
+  "packageManager": "pnpm@9.0.0"
+}
+```
+
+</Tab>
+
+</PackageManagerTabs>
+
+### Step 6: Run you package manager's install command
+
+Update your lockfile by running your installation command.
+
+<PackageManagerTabs>
+
+<Tab>
+
+```bash title="Terminal"
+npm install
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+yarn install
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+pnpm install
+```
+
+</Tab>
+
+</PackageManagerTabs>
+
+Once you've done this, you should see a lockfile diff, indicating that the package has been added to the package manager's workspace.
+
+### Step 7: Install Turborepo
+
+Add Turborepo to the root `package.json` of the workspace.
+
+<PackageManagerTabs>
+
+<Tab>
+
+```bash title="Terminal"
+npm install turbo --save-dev
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+ yarn add turbo --save-dev --ignore-workspace-root-check
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+pnpm install turbo --save-dev --workspace-root
+```
+
+</Tab>
+
+</PackageManagerTabs>
+
+You can also optionally install `turbo` globally for added convenience when working with Turborepo.
+
+<PackageManagerTabs>
+
+<Tab>
+
+```bash title="Terminal"
+npm install turbo --global
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+yarn global add turbo
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+pnpm install turbo --global
+```
+
+</Tab>
+
+</PackageManagerTabs>
+
+### Step 8: Add a `turbo.json`
+
+Create a `turbo.json` at the root to register your tasks and describe their task dependencies.
+
+```json title="./turbo.json"
+{
+  "tasks": {
+    "build": {
+      "dependsOn": ["^build"],
+      "outputs": [".next/**", "!.next/cache/**"]
+    },
+    "dev": {
+      "cache": false,
+      "persistent": true
+    }
+  }
+}
+```
+
+### Step 9: Run `turbo build`
+
+Build the application with Turborepo. Using global `turbo`, this would be `turbo build`. You can also run the command through your package manager:
+
+<PackageManagerTabs>
+
+<Tab>
+
+```bash title="Terminal"
+npx turbo run build
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+ yarn dlx turbo build
+```
+
+</Tab>
+
+<Tab>
+
+```bash title="Terminal"
+pnpm exec turbo build
+```
+
+</Tab>
+
+</PackageManagerTabs>
+
+### Step 10: Enable Remote Caching (optional)
+
+By default, Turborepo will connect to the free-to-use Vercel Remote Cache when you run:
+
+```bash title="Terminal"
+turbo login
+turbo link
+```
+
+You may also configure a self-hosted Remote Cache, which does not require a license or any other fees.
+
+## Advanced migration considerations
+
+While the migration guide above is a good starting point, the breadth of possibilities and capabilities of monorepos means that its difficult to create generalized instructions for all cases. Below, we’ve listed some common next steps that you may be thinking about.
+
+### Creating shared packages
```

Comment:
Do we want to discuss "unhoisting" the external deps from the root and moving them to the packages that use them? Without doing this cache hit rate will be pretty horrid.

Reply:
YES, I never circled back for that. Thank you for reminding me.